### DISCUSSION TITLE
What would you do different, if you could start from scratch?

### DISCUSSION THREAD

**Discussion by guettli** [tox-dev/tox#2230] [General]
Dear tox developers. Imagine tox does not exist. What would you do different, if you could start from scratch? 

Would you use containers?

**Comment by jugmac00** [comment]
Actually, there is a complete rewrite in progress - see 
- https://tox.readthedocs.io/en/rewrite/
- https://github.com/tox-dev/tox/tree/rewrite
- https://github.com/tox-dev/tox/blob/rewrite/src/new.md

@gaborbernat also gave a lightning talk at Pycon US https://youtu.be/5zEn3Jta2Dg?t=391

Though, the rewrite is mostly compatible with tox3.

**Reply by gaborbernat** [reply]
Definitely no to containers ☺️ an option to use containers yes, but the default? No.
